### PR TITLE
fix: dont panic on indexing slicing

### PR DESCRIPTION
**Pull request by Manuthor**
- apply on crates `kmip` and `server` these 2 clippy-lints on indexing slicing:
```
    clippy::indexing_slicing,
    clippy::missing_asserts_for_indexing,
```

### CODE REVIEW

File: crate/kmip/src/kmip/ttlv/deserializer.rs [hunk 1]
```diff
@@ -656,7 +764,15 @@ impl<'de, 'a> de::Deserializer<'de> for &'a mut TtlvDeserializer<'de> {
...
-                let value = &self.get_structure()?[self.index - 1].value;
+                let value = &self
+                    .get_structure()?
+                    .get(self.index - 1)
+                    .ok_or_else(|| {
+                        TtlvError::custom(
+                            "deserialize_identifier: Indexing slicing failed for StructureValue",
+                        )
+                    })?
+                    .value;
```
Comment (ccorsin) [inline crate/kmip/src/kmip/ttlv/deserializer.rs:775]: this repeats a lot - no refacto ?

File: crate/kmip/src/kmip/ttlv/deserializer.rs [hunk 2]
```diff
@@ -487,7 +571,15 @@ impl<'de, 'a> de::Deserializer<'de> for &'a mut TtlvDeserializer<'de> {
                     .to_owned(),
             )),
             Deserializing::StructureValue => {
-                let value = &self.get_structure()?[self.index - 1].value;
+                let value = &self
+                    .get_structure()?
+                    .get(self.index - 1)
+                    .ok_or_else(|| {
+                        TtlvError::custom(
+                            "deserialize_seq: Indexing slicing failed for StructureTag",
```
[the remark this answers is not in the document]
Comment (Manuthor) [reply crate/kmip/src/kmip/ttlv/deserializer.rs]: OK you're right, I will consider those indexing slicing safe.

File: crate/kmip/src/kmip/ttlv/deserializer.rs [hunk 3]
```diff
@@ -563,7 +655,15 @@ impl<'de, 'a> de::Deserializer<'de> for &'a mut TtlvDeserializer<'de> {
     {
         match &self.deserializing {
             Deserializing::StructureValue => {
-                let value = &self.get_structure()?[self.index - 1].value;
+                let value = &self
+                    .get_structure()?
+                    .get(self.index - 1)
+                    .ok_or_else(|| {
+                        TtlvError::custom(
+                            "deserialize_map: Indexing slicing failed for StructureTag",
```
[the remark this answers is not in the document]
Comment (Manuthor) [reply crate/kmip/src/kmip/ttlv/deserializer.rs]: revert changes

File: crate/kmip/src/kmip/ttlv/deserializer.rs (hunk 1 above)
Comment (Manuthor) [reply crate/kmip/src/kmip/ttlv/deserializer.rs]: revert changes

**Pull request merged**